examples/persistent_state.rs

```diff
@@ -0,0 +1,79 @@
+extern crate libsystemd;
+
+use libsystemd::activation;
+use libsystemd::daemon::{self, NotifyState};
+use std::io::{Read, Seek, Write};
+use std::os::unix::prelude::{AsRawFd, IntoRawFd};
+
+use memfd::Memfd;
+
+/*
+cargo build --example persistent_state
+systemd-run --user -p RestartForceExitStatus=2 -p SuccessExitStatus=2 -p FileDescriptorStoreMax=1 --wait ./target/debug/examples/persistent_state
+journalctl --user -xeu <unitname>.service
+*/
+
+fn create_and_store_persistent_state() -> Memfd {
+    let mfd = memfd::MemfdOptions::default()
+        .create("persistent-state")
+        .unwrap();
+
+    let nss = [
+        NotifyState::Fdname("persistent-state".to_owned()),
+        NotifyState::Fdstore,
+    ];
+    daemon::notify_with_fds(false, &nss, &[mfd.as_raw_fd()]).expect("notify failed");
+    // Set initial state to 0
+    let state = [0u8, 1];
+    mfd.as_file().set_len(state.len() as u64).unwrap();
+    mfd.as_file().write(&state).unwrap();
+    mfd.as_file().rewind().unwrap();
+    mfd
+}
+
+fn main() {
```

Comment:
I'd consider changing this to a `run() -> Result<i32>` so that you can cut down the amount of unwrap/expect, and leave a single process-exit to `main()`.